sources/TileDB.CSharp/Enums.cs

```diff
@@ -1157,16 +1162,11 @@ public static DataType TypeToDataType(Type t)
             }
             else
             {
-#pragma warning disable CS0618 // Type or member is obsolete
-                return DataType.Any;
-#pragma warning restore CS0618 // Type or member is obsolete
+                ThrowHelpers.ThrowTypeNotSupported();
+                return default; // unreachable
```

Comment:
Thoughts on this? Returning `Any` on an unsupported type can be misleading since `Any` is obsolete and there is no clear documentation on what to do with it.

Reply:
What is `default` here?

Reply:
I think throwing is fine, if you mean `TILEDB_ANY`, there's definitely no way to translate that.

Reply:
In general `default` is equal to `null` on reference types, zero on primitives and all fields recursively set to `default` in value types.

In this case I added it to satisfy the compiler. It won't be returned because the above function will throw (we do it in separate method for performance reasons).